### ISSUE TITLE
Toggle between layout cause a lot of white squares on screen

### ISSUE THREAD

**Issue by zk4** [kovidgoyal/kitty#2829]
**Describe the bug**
Toggle between layout cause a lot of  white squares on screen. 
  
**Screenshots**
![2020-07-03](https://user-images.githubusercontent.com/996674/86459737-da464a00-bd59-11ea-9c29-f4e48cede58a.gif)

**Environment details**
Mac:  OS 10.14.6
Kitty: 0.17.4

```
[185 18:21:50.072073] Symbol map: 3400-4DB5 20000-2A6D6 2A700-2B734 2B740-2B81D 2B820-2CEA1 2CEB0-2EBE0 2F00-2FD5 2E80-2EF3 F900-FAD9 2F800-2FA1D E815-E86F E400-E5E8 E600-E6CF 31C0-31E3 2FF0-2FFB 3105-312F 31A0-31BA 3007 FangSong is invalid, ignoring

Config options different from defaults:
active_border_color          Color(red=255, green=255, blue=0)
active_tab_background        Color(red=59, green=60, blue=53)
active_tab_foreground        Color(red=238, green=38, blue=109)
allow_remote_control         y
background                   Color(red=59, green=60, blue=53)
color0                       Color(red=59, green=60, blue=53)
color1                       Color(red=248, green=37, blue=112)
color10                      Color(red=166, green=225, blue=45)
color11                      Color(red=228, green=219, blue=115)
color12                      Color(red=252, green=150, blue=31)
color13                      Color(red=174, green=129, blue=255)
color14                      Color(red=102, green=217, blue=238)
color15                      Color(red=253, green=255, blue=241)
color2                       Color(red=166, green=225, blue=45)
color3                       Color(red=228, green=219, blue=115)
color4                       Color(red=252, green=150, blue=31)
color5                       Color(red=174, green=129, blue=255)
color6                       Color(red=102, green=217, blue=238)
color7                       Color(red=253, green=255, blue=241)
color8                       Color(red=110, green=112, blue=102)
color9                       Color(red=248, green=37, blue=112)
copy_on_select               clipboard
cursor                       Color(red=253, green=255, blue=241)
cursor_text_color            Color(red=0, green=0, blue=0)
dim_opacity                  1.0
enabled_layouts              ['horizontal', 'vertical', 'grid', 'tall', 'stack', 'fat']
font_family                  RobotoMonoForPowerline-Light
font_size                    16.0
foreground                   Color(red=253, green=255, blue=241)
inactive_border_color        Color(red=17, green=17, blue=17)
inactive_tab_background      Color(red=34, green=34, blue=34)
inactive_tab_foreground      Color(red=238, green=238, blue=238)
listen_on                    'unix:/tmp/mykitty'
macos_custom_beam_cursor     True
macos_thicken_font           0.5
macos_traditional_fullscreen True
open_url_modifiers           8
selection_background         Color(red=253, green=255, blue=241)
selection_foreground         Color(red=59, green=60, blue=53)
symbol_map                   {(19968, 40943): 'FangSong'}
tab_bar_edge                 1
tab_bar_style                separator
tab_separator                 | 
tab_title_template           {index}:{title}
url_color                    Color(red=255, green=0, blue=0)
Added shortcuts:
	control+h KeyAction(func='kitten', args=['pass_keys.py', 'neighboring_window left   ctrl+h'])
	control+j KeyAction(func='kitten', args=['pass_keys.py', 'neighboring_window bottom ctrl+j'])
	control+k KeyAction(func='kitten', args=['pass_keys.py', 'neighboring_window top    ctrl+k'])
	control+l KeyAction(func='kitten', args=['pass_keys.py', 'neighboring_window right  ctrl+l'])
	control+alt+s KeyAction(func='goto_layout', args=['stack'])
	control+alt+t KeyAction(func='goto_layout', args=['tall'])
	super+d KeyAction(func='new_window_with_cwd', args=())
	super+e KeyAction(func='kitten', args=['mykitten.py', "':SwitchBuffer\\r'"])
	super+g KeyAction(func='kitten', args=['mykitten.py', "'n'"])
	super+k KeyAction(func='clear_terminal', args=['reset', True])
	shift+super+minus KeyAction(func='change_font_size', args=(True, '-', 2.0))
	shift+super+equal KeyAction(func='change_font_size', args=(True, '+', 2.0))
	shift+super+f KeyAction(func='kitten', args=['mykitten2.py', '"fcmd \\r"'])
	shift+super+o KeyAction(func='kitten', args=['mykitten.py', "':LeaderfFunction\\r'"])
	shift+super+r KeyAction(func='kitten', args=['mykitten2.py', '"ranger \\r"'])
	shift+super+v KeyAction(func='kitten', args=['mykitten2.py', '"nvim .\\r"'])
	shift+super+enter KeyAction(func='kitten', args=['zoom_toggle.py'])
	control+super+h KeyAction(func='neighboring_window', args=['left'])
	control+super+j KeyAction(func='neighboring_window', args=['bottom'])
	control+super+k KeyAction(func='neighboring_window', args=['top'])
	control+super+l KeyAction(func='neighboring_window', args=['right'])
	control+super+n KeyAction(func='scroll_line_down', args=())
	control+super+p KeyAction(func='scroll_line_up', args=())
	shift+control+super+t KeyAction(func='set_tab_title', args=())
	alt+super+b KeyAction(func='kitten', args=['mykitten.py', "':CocFormat\\r'"])
	shift+alt+super+left_bracket KeyAction(func='move_tab_backward', args=())
	shift+alt+super+right_bracket KeyAction(func='move_tab_forward', args=())
Changed shortcuts:
	shift+control+f10 KeyAction(func='kitty_shell', args=['window'])
	super+1 KeyAction(func='move_window_backward', args=())
	super+2 KeyAction(func='move_window_forward', args=())
	super+3 KeyAction(func='start_resizing_window', args=())
	super+t KeyAction(func='new_tab_with_cwd', args=['!neighbor'])
	super+w KeyAction(func='close_window', args=())
	shift+super+d KeyAction(func='next_layout', args=())

```

**Additional context**
key binding
``` conf
map cmd+shift+enter kitten zoom_toggle.py
```
the script:
zoom_toggle.py
``` py
def main(args):
   pass

def handle_result(args, answer, target_window_id, boss):
   tab = boss.active_tab
   if tab is not None:
      if tab.current_layout.name == 'stack':
         tab.last_used_layout()
      else:
         tab.goto_layout('stack')

handle_result.no_ui = True

```

**Comment by trygveaa** [comment]
Seems like a duplicate of #2747. Fixed in v0.18.0.

**Comment by zk4** [comment]
@trygveaa  Thanks a lot! That`s the same bug. Issue closed.
